src/datautils/wellboreItemShapeGenerator.ts

```diff
@@ -57,56 +52,93 @@ export const cementDiameterChangeDepths = (
 ): number[] => {
   const topOfCement = cement.toc;
 
-  const diameterChangeDepths = merge(
-    diameterIntervals.map((d) => [
-      d.start - EPSILON, // +- 0.0001 to find diameter right before object
-      d.start,
-      d.end,
-      d.end + EPSILON, // +- 0.0001 to find diameter right after object
-    ]),
-  ).filter((d) => d >= topOfCement && d <= bottomOfCement) as number[]; // trim
+  const diameterChangeDepths = diameterIntervals.flatMap((d) => [d.start, d.end]);
+  const trimmedChangedDepths = diameterChangeDepths.filter((d) => d >= topOfCement && d <= bottomOfCement) as number[]; // trim
 
-  diameterChangeDepths.push(topOfCement);
-  diameterChangeDepths.push(bottomOfCement);
+  trimmedChangedDepths.push(topOfCement);
+  trimmedChangedDepths.push(bottomOfCement);
 
-  const uniqDepths = uniq(diameterChangeDepths);
+  const uniqDepths = uniq(trimmedChangedDepths);
 
   return uniqDepths.sort((a: number, b: number) => a - b);
 };
 
-export const calculateCementDiameter =
-  (innerCasing: Casing[], nonAttachedCasings: Casing[], holes: HoleSize[]) =>
-  (
-    depth: number,
-  ): {
-    md: number;
-    innerDiameter: number;
-    outerDiameter: number;
-  } => {
-    const defaultCementWidth = 100; // Default to flow cement outside to show error in data
-
-    const innerCasingAtDepth = innerCasing.find((casing) => casing.start <= depth && casing.end >= depth);
-    const innerDiameter = innerCasingAtDepth ? innerCasingAtDepth.diameter : 0;
-
-    const outerCasings = nonAttachedCasings.filter((casing) => casing.innerDiameter > innerDiameter);
-    const holeAtDepth = holes.find((hole) => hole.start <= depth && hole.end >= depth && hole.diameter > innerDiameter);
-    const outerCasingAtDepth = outerCasings
-      .filter((d) => d)
-      .sort((a, b) => a.innerDiameter - b.innerDiameter) // ascending
-      .find((casing) => casing.start <= depth && casing.end >= depth && casing.diameter > innerDiameter);
-
-    let outerDiameter;
-    if (outerCasingAtDepth) {
-      outerDiameter = outerCasingAtDepth.innerDiameter;
-    } else if (holeAtDepth) {
-      outerDiameter = holeAtDepth.diameter - HOLE_OUTLINE;
-    } else {
-      outerDiameter = defaultCementWidth;
-    }
+export const findCementOuterDiameterAtDepth = (innerCasing: Casing[], nonAttachedCasings: Casing[], holes: HoleSize[], depth: number): number => {
+  const defaultCementWidth = 100; // Default to flow cement outside to show error in data
+
+  const innerCasingAtDepth = innerCasing.find((casing) => casing.start <= depth && casing.end >= depth);
+  const innerDiameter = innerCasingAtDepth ? innerCasingAtDepth.diameter : 0;
+
+  const outerCasings = nonAttachedCasings.filter((casing) => casing.innerDiameter > innerDiameter);
+  const holeAtDepth = holes.find((hole) => hole.start <= depth && hole.end >= depth && hole.diameter > innerDiameter);
+  const outerCasingAtDepth = outerCasings
+    .filter((d) => d)
```

Comment:
Don't think this truthy filter is no longer necessary, as the filter on L72 will only return casings.